lib/fluent/plugin/in_prometheus.rb

```diff
@@ -184,7 +190,16 @@ def start_webrick
     end
 
     def all_metrics
-      [200, { 'Content-Type' => ::Prometheus::Client::Formats::Text::CONTENT_TYPE }, ::Prometheus::Client::Formats::Text.marshal(@registry)]
+      body = nil
+      case @content_encoding
+      when 'gzip'
```

Comment:
```suggestion
      when :gzip
```

(If you change the type of it to `:enum`.)